### PR TITLE
Changes the Blaze Dashboard path to the new format

### PR DESCRIPTION
**Pull request by sbarbosa**
Some time ago (https://github.com/Automattic/wp-calypso/pull/79838), we updated the navigation path of Blaze to follow Calypso standards (we moved the site fragment to the end of the path). We currently have some compatibility redirections to make the change compatible with Jetpack.

This PR will update the paths to the Blaze Dashboard to the new path format.

Old Format: `tools.php?page=advertising#!/advertising/{SITE_SLUG}/posts/promote/post-{POST_ID}`
New Format: `tools.php?page=advertising#!/advertising/posts/promote/post-{POST_ID}/{SITE_SLUG}`

## Proposed changes:
* Updates the path used in the post row action and in the Post publish widget.
* Changes a typo in the Blaze configuration initial state. The Blaze Dashboard app is already compatible with both names (`intial_state` and `initial_state`).

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Go to the Post page and verify that the link generated for the "Promote with Blaze" link is correct (it has the site name at the end)
* Click the "Promote with Blaze" link for a post and verify that you land correctly in the Create campaign widget for the post you just selected
* Go to the Post page again, but now create a new post and publish it
* In the Post publish widget, click on the "Blaze this post" link
* Just as before, verify that you landed in the correct Blaze page for the published post.

### CODE REVIEW

File: projects/packages/blaze/changelog/update-blaze-dashboard-urls [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
```
Comment (jeherve) [inline projects/packages/blaze/changelog/update-blaze-dashboard-urls]: Maybe this should be a `minor` bump instead, what do you think?

```suggestion
Significance: minor
```

**Review by jeherve — COMMENTED** [review]
You'll need to bump the version with `tools/fixup-project-versions.sh` so we can merge this.

**Review by therocket-gr — DISMISSED** [review]
tested locally and i confirm that the change in the url is working as described
Screenshot 2024-01-09 at 13 49 36

Changes LGTM (but i guess we need to address what @jeherve has noted above

**Comment by sbarbosa** [comment]
Thanks @jeherve, @therocket-gr! 
I applied the suggestions and updated the PR with trunk.

**Review by therocket-gr — APPROVED** [review]
LGTM

**Review by jeherve — APPROVED** [review]

**Pull request merged**